waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -782,12 +782,41 @@
    :periodSeconds health-check-interval-secs
    :timeoutSeconds 1})
 
+(defn envoy-sidecar-enabled?
+  "Returns true if the envoy sidecar is enabled, and false otherwise"
+  [_ _ {:strs [env]} _]
+  (let [reverse-proxy-flag "REVERSE_PROXY"]
+    (contains? env reverse-proxy-flag)))
+
+(defn attach-envoy-sidecar
+  "Attaches envoy sidecar to replicaset"
+  [replicaset reverse-proxy service-description base-env service-port port0]
+  (update-in replicaset
+    [:spec :template :spec :containers]
+    conj
+    (let [{:keys [cmd resources image]} reverse-proxy
+          user-env (:env service-description)
+          env-vector (-> user-env
+                         (merge base-env)
+                         (assoc "PORT0" (str port0) "SERVICE_PORT" (str service-port)))
+          env (into []
```

Comment:
`env` is a vector, maybe we should flip the names.

Reply:
Done